### ISSUE TITLE
Fixer le problème de soumission sur TOP-MENELAS

### ISSUE THREAD

**Issue by jonquet** [sifrproject/bioportal_web_ui#9]
http://bioportal.lirmm.fr/ontologies/TOP-MENELAS/?p=classes&conceptid=root
Probleme with class

Et on a ca:
http://data.bioportal.lirmm.fr/ontologies/TOP-MENELAS/classes
"no submission"

Pourtant il y a maintenant une soumission:
http://data.bioportal.lirmm.fr/ontologies/TOP-MENELAS/latest_submission

**Comment by vemonet** [comment]
Problème résolu. Il suffisait de relancer le parsing

Logs de l'erreur:
`D, [2016-11-16T16:05:11.146344 #23372] DEBUG -- : Starting parsing for http://data.bioontology.org/ontologies/TOP-MENELAS/submissions/1
D, [2016-11-16T16:05:11.146443 #23372] DEBUG -- : Pull location found, but no file in the upload file path. Retrying download.`

Et rien après "Retrying download". Donc c'est qu'il y a un petit blocage quelque part au moment de relancer le parsing.

**A investiguer si on constate que ça se reproduit**

**Comment by vemonet** [comment]
Je confirme, ce problème semble s'être reproduit avec ONTOPNEUMO

**Comment by jonquet** [comment]
Probleme réglé aussi sur ONTOPNEUMO visiblement. Je ferme le tracker et te laisse expliquer en fermant le tracker #11

**Comment by vemonet** [comment]
Quand on ajoute une submission dans plusieurs cas (il semble que ce soit en particulier pour les ontologies qu'on essaie de puller du NCBO, je test sur Stage) le parsing arrive jusqu'à cette ligne :
https://github.com/sifrproject/ncbo_cron/blob/master/lib/ncbo_cron/ontology_submission_parser.rb#L168

Puis plus rien

Il faut alors relancer le parsing manuellement (/srv/ncbo/ncbo_cron/bin/ncbo_ontology_process -o ONTOPNEUMO) pour que ça fonctionne...

Note: on a jamais touché quoi que ce soit lié au ddl d'ontologie. C'est un bug à génération spontanée apparemment..

**Comment by vemonet** [comment]
Réponse de John Graybeal : 

"Just throwing an idea out there before the end of our day, since we have a long turnaround time across the big water. 

This feels like a permissions problem. If it's running OK manually, maybe you have a permission that the automated process doesn't.

And now that I've looked at the github code, it seems even more likely.

And the jira issue reinforces this: "Newly upload files have wrong permissions". It appears files in the repository directory are created with the wrong permissions, but we've never figured out why.  

Alex fixed it as a one-off back in 2014: "I manually reset file permissions for repository directory; however, it is unclear why those files are created with wrong permissions". The debug code was put in to try to determine when/why the permissions are failing; initial indications are the file has the wrong permission, maybe 0 instead of something readable.  But not sure how it gets that way.

Ah, the last comment is in 2015, Alex writes:

> is it happening because dante sets umask 0000 when forking?
>   https://github.com/nesquena/dante/blob/master/lib/dante/runner.rb#L91 

So, short term, try resetting the permissions of files in the source directory to allow them to be read, and make sure the destination directory is writeable. 

As far as the root cause of this problem, maybe we've learned something since 2015, I'll let the experts weigh in on that possibility."
